dotnet-engine/UnleashEngine.Tests/UnleashEngineTest.cs

```diff
@@ -14,11 +14,50 @@ public class Tests
         PropertyNamingPolicy = JsonNamingPolicy.CamelCase
     };
 
+    [Test] 
+    public void MassTestMemoryUsage() {
+        // Arrange
+        var basePath = Path.Combine("..", "..", "..", "..", "..", "..", "client-specification", "specifications");
+        var suitePath = Path.Combine(basePath, "01-simple-examples.json");
+        var suiteData = JObject.Parse(File.ReadAllText(suitePath));
+
+        var unleashEngine = new UnleashEngine();
+        //unleashEngine.TakeState(suiteData["state"].ToString());
+
+        var runTestFor = (Action lambda, string process) => {
+
+            // Baseline / warm up
+            for (var i = 0; i < 1000000; i++) {
+                lambda();
+            }
+            GC.Collect();
+
+            var baseline = GC.GetTotalMemory(true);
+
+            // Act
+            for (var i = 0; i < 1000000; i++) {
+                lambda();
+            }
+            GC.Collect();
+
+            var memoryTotal = GC.GetTotalMemory(true);
+
+            // Assert
+            var diff = memoryTotal - baseline;
+            Assert.LessOrEqual(diff, 524288, process + " has a likely memory leak. Diff: " + diff + " bytes");

```

Comment:
Is this in bytes? So this is roughly half a meg?

Reply:
Yeah my brain's trapped in the pre MB/MiB times - so this is half a MiB. I'll change to half a MB.
But yes. It's heavily influenced by the Ruby implementation - the real diff appears to be around a tenth of that (or a hundredth when the engine actually has state first)

Reply:
Let's be more aggressive then! If .NET is better than Ruby with memory (really, really not surprising), let's get this closer to the real number!